components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -467,14 +469,31 @@ class GeckoEngineSession(
             val visitType = if (isReload) {
                 VisitType.RELOAD
             } else {
-                if (flags and GeckoSession.HistoryDelegate.VISIT_REDIRECT_SOURCE_PERMANENT != 0) {
+                // Note the difference between `VISIT_REDIRECT_PERMANENT`,
+                // `VISIT_REDIRECT_TEMPORARY`, `VISIT_REDIRECT_SOURCE`, and
+                // `VISIT_REDIRECT_SOURCE_PERMANENT`.
+                //
+                // The former two indicate if the visited page is the *target*
+                // of a redirect; that is, another page redirected to it.
+                //
+                // The latter two indicate if the visited page is the *source*
+                // of a redirect: it's redirecting to another page, because the
+                // server returned an HTTP 3xy status code.
+                if (flags and GeckoSession.HistoryDelegate.VISIT_REDIRECT_PERMANENT != 0) {
```

Comment:
nit: I'm assuming these check works correctly as 
((flags and GeckoSession.HistoryDelegate.VISIT_REDIRECT_PERMANENT) != 0)
But I can't confirm that looking at Kotlin language docs.  Can we explicitly put the brackets around so its clear for readers?